core/tx_pool.go

```diff
@@ -437,6 +447,22 @@ func (pool *TxPool) loop() {
 					queuedEvictionMeter.Mark(int64(len(list)))
 				}
 			}
+			// Evict pending transactions
+			for addr := range pool.pending {
+				// Skip local transactions from the eviction mechanism
+				if pool.locals.contains(addr) {
+					continue
+				}
+				// Any non-locals old enough should be removed
+				if time.Since(pool.beats[addr]) > pool.config.Lifetime {
+					list := pool.pending[addr].Flatten()
```

Comment:
Should probably call `pool.pending[addr].txs.flatten()` to avoid copying of the slice

Reply:
True, but this happens rarely and I don't expect it to lead to too much overhead, so I opted for consistency with existing code.